src/ui/popupmenu/popupmenu.rs

```diff
@@ -388,9 +388,11 @@ impl Popupmenu {
                 prev.info.set_visible(false);
                 prev.menu.set_visible(false);
 
-                // Update the `kind` icon with default fg color.
+                // Update the `kind` icon with default fg color if it is an icon.
                 let buf = get_icon_pixbuf(&prev.item.kind, &fg, font_height);
-                prev.kind.set_from_pixbuf(&buf);
+                if items.iter().any(|item| !item.kind.is_unknown())  {
```

Comment:
Here, you can use `state.show_kind` once that is added.

Reply:
@vhakulinen Do you want me to add `show_kind` to the _popupmenu's_ `State` struct, or is there another way to use `state.show_kind` here that doesn't require adding that?

Reply:
Ah, right. The `show_kind` should go into the `LazyLoader`'s state. You'll have to add an function to `LazyLoader` to then get it.